### PR TITLE
Refactor modular

### LINKED ISSUE #71: CLEANUP: Refactor code for modularity
As title says, yo

### PR DESCRIPTION
**Pull request by tiffeli**
- we were doing navigator.geolocation functions a lot so I thought we could definitely abstract that function and be more DRY.

- all the navigator functions are located in utils.js and imported as geoAction in the files.

1. getCurrent = navigator.geolocation.getCurrentPosition

        - takes in a callback that is passed in successfully found coords (object w latitude and longitude)

2. setWatch= navigator.geolocation.watchPosition
        - takes in a callback that is passed in successfully found coords (object w latitude and longitude)

3. clearWatch = navigator.geolocation.clearWatch

        - takes in watchID that you want to clear.
 

**note: remember to save this binding by declaring self = this outside of the function and refer all 'this' as self**. See the refactored functions for example.

- call them by invoking eg:  geoAction.getCurrent()

resolves #71

### CODE REVIEW

**Comment by jake-shasteen** [comment]
./lib/orientation might be a better place for this to live, since all of the tools in there have to do with real world location and direction

**Comment by tiffeli** [comment]
@jake-shasteen moved it! :heavy_check_mark:

File: app/actions/action_dropNewPin.js [hunk 1]
```diff
@@ -51,15 +53,9 @@ export default function getLocationToSave(location, current, pinTitle) {
...
-             getLocationHelper(position.coords, "My Current Location", dispatch);
-          },
-          (error) => {
-            console.error(error);
-          },
-          {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000}
-        );
+        geoAction.getCurrent((loc)=>{
+          getLocationHelper(position.coords, "My Current Location", dispatch);
+        });
```
Comment (jake-shasteen) [inline app/actions/action_dropNewPin.js]: You're referencing a variable `loc` in the anonymous function definition but don't seem to be making use of it here.

**Comment by jake-shasteen** [comment]
Found a bug introduced by a previous PR -- there is an if statement inside Map.js' render() that needs to be moved out to componentWillUpdate() or it will cause the app to crash when putting down a new pin and changing views.

Also, it seems like the 'go to shared pin' function when a friend shares a pin with you is not currently working as intended.

**Comment by tiffeli** [comment]
I'm going to close this one, since I had to merge two PR's and we shouldn't mess up my commit orders.

**Pull request closed without merging**